File: Tanishq Singhai/Sort Array By Parity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nums) {
         // Time Complexity : O(n)
         // Space Complexity : O(1)
        int n=nums.size();
        int i=0, j=n-1;
        while (i<j) {
            if (nums[i]%2 > nums[j]%2) 
                swap(nums[i], nums[j]);
            if(nums[i]%2 == 0) i++;
            if(nums[j]%2 == 1) j--;
        }
        return nums;
    }
};

File: Parth Mathur/SortArrayByParity.java
class Solution {
    public int[] sortArrayByParity(int[] arr) {
        int i=0;
        int j = arr.length-1;
        
        while(i<j)
        {
            if(arr[i] % 2 == 0)
            {
                i++;
            }
            else
            {
                if(arr[j] % 2 != 0)
                {
                    j--;
                }
                else
                {
                int temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
                
                j--;
                i++;  
                }
            }
            
        }
        return arr;
        
    }
}
